mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -415,7 +415,8 @@ final class OnlineLDAOptimizer extends LDAOptimizer with Logging {
       docs: RDD[(Long, Vector)],
       lda: LDA): OnlineLDAOptimizer = {
     this.k = lda.getK
-    this.corpusSize = docs.count()
+    this.docs = docs.filter(_._2.numNonzeros > 0) // filter out empty documents
+    this.corpusSize = this.docs.count()
     this.vocabSize = docs.first()._2.size
```

Comment:
Needs to be `this.docs` too

Reply:
`docs` is assumed to be non-empty, while `this.docs` may be empty. In this case `first()` fails.